### PR TITLE
feat: add `aria-required` attributes to WC form fields

### LINKED ISSUE #44663: Programmatically Indicate Required Fields in Checkout (Using Shortcode)
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

It's important for screen-reading software to get as much information as possible about form fields.

Right now WooCommerce isn't programmatically indicating which fields are required. This means users who use screen reader software can be unaware a field is required.

See WCAG guidelines about required fields: https://www.w3.org/TR/WCAG20-TECHS/ARIA2.html

### Expected behavior

When I use screen reading software to navigate to an input field, it should tell me if the field is required or not.

### Actual behavior

I used screen reading software and it doesn't tell me any of the fields are required.

(🔉 on)

https://github.com/woocommerce/woocommerce/assets/1065372/bbc219f4-8346-4ae6-8214-cfc6ecc1f5f3

### Steps to reproduce

1. Use the classic checkout shortcode on your checkout page (this doesn't affect blocks)
2. Add a product to your cart
3. Go to checkout
4. Use screen reading software (I use [Screen Reader](https://chromewebstore.google.com/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn?pli=1) in Chrome)
5. Navigate to input fields

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): http://woo-core.local
Site address (URL): http://woo-core.local
WC Version: 8.5.2
REST API Version: ✔ 8.5.2
WC Blocks Version: ✔ 11.8.0-dev
Action Scheduler Version: ✔ 3.7.1
Log Directory Writable: ✔
WP Version: 6.4.3
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.16.0
PHP Version: 8.1.23
PHP Post Max Size: 1,000 MB
PHP Time Limit: 1200
PHP Max Input Vars: 4000
cURL Version: 8.4.0
(SecureTransport) LibreSSL/3.3.6

SUHOSIN Installed: –
MySQL Version: 8.0.16
Max Upload Size: 300 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 8.5.2
WC Database Prefix: wp_
Total Database Size: 4.88MB
Database Data Size: 3.19MB
Database Index Size: 1.69MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 2.06MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_addresses: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_orders: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_orders_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 11
page: 7
post: 2
product: 18
revision: 3
shop_order_placehold: 1
wp_global_styles: 1
wp_navigation: 1
wp_template: 2

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (1) ###

WooCommerce: by Automattic – 8.5.2

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to Woo.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature screen enabled: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### WC Pages ###

Shop base: #7 - /shop/
Cart: #8 - /cart/
Checkout: #9 - /checkout/
My account: #10 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 4.5.4
Author URL: https://woo.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-variation-management
product-virtual-downloadable
product-external-affiliate
product-grouped
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: customize-store
minified-js
new-product-management-experience
product-linked
settings
async-product-editor-category-field

Daily Cron: ✔ Next scheduled: 2024-02-15 21:22:38 +00:00
Options: ✔
Notes: 75
Onboarding: completed

### Action Scheduler ###

Complete: 30
Oldest: 2024-02-05 21:23:41 +0000
Newest: 2024-02-15 20:24:52 +0000

Failed: 1
Oldest: 2024-02-05 21:24:38 +0000
Newest: 2024-02-05 21:24:38 +0000

Pending: 1
Oldest: 2024-02-16 20:21:58 +0000
Newest: 2024-02-16 20:21:58 +0000

### Status report information ###

Generated at: 2024-02-15 20:33:47 +00:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by frosso**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Adding the `aria-required="true"` attribute to required form inputs.
I chose to add the `aria-required` attribute rather than the `required` attribute to avoid backwards-compatibility issues.
When the `required` attribute is added, the form submission is intercepted and prevented by the browser.
By adding `aria-required`, instead`, we add information to Screen Reader users without affecting the form's behavior.

I ensured that the `aria-required` attribute was added to all input types, excluding `hidden`.
The `aria-required` attribute is added to all inputs of type `radio`, [per HTML specs](https://html.spec.whatwg.org/multipage/input.html#the-required-attribute):
> To avoid confusion as to whether a [radio button group](https://html.spec.whatwg.org/multipage/input.html#radio-button-group) is required or not, authors are encouraged to specify the attribute on all the radio buttons in a group. 

Closes https://github.com/woocommerce/woocommerce/issues/44663

### How to test the changes in this Pull Request:

- (optional) Add the following snippet to your installation to test the radio inputs(you can use [the Code Snippets plugin](https://wordpress.org/plugins/code-snippets/), for example)
```php
add_action( 'woocommerce_after_checkout_billing_form', 'checkout_test_radio_options' );
function checkout_test_radio_options() {
	echo "<div>";
	echo sprintf( '<p>%s</p>', __( "Test radio input" ) );
    woocommerce_form_field( 'test_radio',  array(
		'required'  => true,
        'type'      => 'radio',
        'class'     => array( 'form-row-wide' ),
        'options'   => array(
            '1' => 'Option 1',
            '2' => 'Option 2',
        ),
    ), '1' );
	echo "</div>";
}
```
- Add a product to the cart
- Go to the shortcode checkout page
- The checkout page should display a wide range of inputs. All required inputs (excluding the "hidden" ones) should have an `aria-required="true"` attribute

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
feat: add `aria-required` attributes to WC form fields
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Comment by frosso** [comment]
Assigned to Rubik because the change affects checkout form fields on classic checkout. Feel free to reassign if incorrect!

File: plugins/woocommerce/tests/legacy/unit-tests/templates/functions.php
Comment (alexflorisca) [inline plugins/woocommerce/tests/legacy/unit-tests/templates/functions.php]: This should make the test pass
```suggestion
				'id'       => 'test_radio',
```

**Review by alexflorisca — COMMENTED** [review]
The code looks good and the `aria-required` attribute is added for all inputs except hidden ones. However for the `checkbox` input type, the `aria-label` is added onto the `label` rather than the `input`. 

The test you added is also failing, I added a suggestion below that should fix it. 

Nice work, ping me again when you see this and push changes and I'll have another look 👍

File: plugins/woocommerce/tests/legacy/unit-tests/templates/functions.php
Comment (frosso) [reply plugins/woocommerce/tests/legacy/unit-tests/templates/functions.php]: Ope, thank you!

**Comment by frosso** [comment]
Thank you @alexflorisca ! I fixed the linting issues & added a new test for checkboxes (ensuring the `aria-attribute` is added to the input).
Thanks again for your time!

**Review by opr — CHANGES_REQUESTED** [review]
This is a really nice clean solution, and I like that you opted for `aria-required`.

The changes you made here won't apply to country or state select fields as these are converted into select2 instances. For these you would need to update `selectWoo.full.js`.

https://github.com/woocommerce/woocommerce/commit/bbabe8a8efab6a74633062b6955f634c74c559f1 shows how this would look, though you need to change the `country-select.js` code to check for `aria-required` instead of `required`

Also in 718bd3c9800efa9a63e680862ec5f421c559caab I made it so the asterisk is not read by screen readers. What do you think about applying this change too?

**Comment by frosso** [comment]
Thank you for the feedback, @opr !

> The changes you made here won't apply to country or state select fields as these are converted into select2 instances. For these you would need to update `selectWoo.full.js`.

> https://github.com/woocommerce/woocommerce/commit/bbabe8a8efab6a74633062b6955f634c74c559f1 shows how this would look, though you need to change the `country-select.js` code to check for `aria-required` instead of `required`

With this approach, were you able to make the Screen Reader (SR) software announce the field as "required"?
With the changes implemented (as a note - slightly differently because the `$this.attr('required')` doesn't return a boolean but a string & applying the `required` attribute on the `role="combobox"` element), I am still not able to have any software announce the field as required.
I tried with VoiceOver on iOS/Mac OS, Talkback on Android Chrome & Brave, [Screen Reader in Chrome](https://chromewebstore.google.com/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn?pli=1).

> Also in 718bd3c I made it so the asterisk is not read by screen readers. What do you think about applying this change too?

I'd like to hear your thoughts on the issue I mentioned above above, first.
Since some browsers & SR software combinations don't announce required fields (e.g.: [Talkback on Android w/ Google Chrome](https://www.tpgi.com/required-attribute-requirements/)), I think the "required" `abbr` (albeit redundant in some other cases) provides a useful hint to users that otherwise wouldn't get the hint.

If we removed the `required` abbr, do you think we should also remove the `(optional)` hint on optional fields at checkout?

FWIW, I also noticed that to be really consistent, we'd need to add a `aria-hidden` attribute [on `address-i18n.js` as well](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/legacy/js/frontend/address-i18n.js#L17-L21).

Since both issues go hand-in-hand, I was instead wondering if we should keep the "required" `abbr` tag to ensure that required `select2` fields can be announced by SR software.

**Comment by opr** [comment]
@frosso 

> With this approach, were you able to make the Screen Reader (SR) software announce the field as "required"?
With the changes implemented (as a note - slightly differently because the $this.attr('required') doesn't return a boolean but a string & applying the required attribute on the role="combobox" element), I am still not able to have any software announce the field as required.
I tried with VoiceOver on iOS/Mac OS, Talkback on Android Chrome & Brave, [Screen Reader in Chrome](https://chromewebstore.google.com/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn?pli=1).

Yes, I was, I just implemented my changes on your branch. I tried with VoiceOver and Screen reader in Chrome and both worked for me! Can you check out the [`origin/try/country-screen-reader` branch](https://github.com/woocommerce/woocommerce/tree/try/country-screen-reader) and see if it works there?

> Since some browsers & SR software combinations don't announce required fields (e.g.: [Talkback on Android w/ Google Chrome](https://www.tpgi.com/required-attribute-requirements/)), I think the "required" abbr (albeit redundant in some other cases) provides a useful hint to users that otherwise wouldn't get the hint.

I didn't realise that some wouldn't announce the field as required! I agree the hints are probably useful then, let's leave them!

**Comment by frosso** [comment]
> Yes, I was, I just implemented my changes on your branch. I tried with VoiceOver and Screen reader in Chrome and both worked for me! Can you check out the [origin/try/country-screen-reader branch](https://github.com/woocommerce/woocommerce/tree/try/country-screen-reader?rgh-link-date=2024-06-17T13%3A39%3A37Z) and see if it works there?

I applied those changes on my local environment.
I also added a required `select` on checkout with label "Test required input".

I couldn't record Talkback on Android, due to some permissions issue. However, this is the transcribed result on Brave browser:

<details>
<summary>Brave Browser transcription</summary>

"Test required input" label:
> Test required input

"Test required input" input:
> Collapsed, required, test required input, menu popup button, double tap to activate

"First name" label:
> First name

"First name" input:
> Required, edit box, first name, double tap to edit text

"Country / region" label
> Country / region

"Country / region" selectWoo input
> Collapsed, United States US, dropdown list, country / region, double tap to change
</details>

On Android with Talkback & Google Chrome, all inputs (and labels) are not announced as required.

Mac OS w/ VoiceOver on Safari - VoiceOver doesn't seem to announce the "selectWoo" inputs as required. It's skipping the "required" `abbr` tag and announcing other native fields as `required`.

https://github.com/woocommerce/woocommerce/assets/273592/00292fa4-dd3b-48a2-931c-3070e9fa5f40

So with that branch's changes the "Country / Region" was never announced as required during my testing.

**Comment by opr** [comment]
@frosso very strange! It looks like that select isn't styled correctly , I'm wondering if it really is being picked up by select2 on your site?

Does it have `aria-required="true"` on it?

When I inspect it I see this, what do you see?

image

**Comment by frosso** [comment]
> Does it have aria-required="true" on it?
> When I inspect it I see this, what do you see?

I have a similar output
Screenshot 2024-06-17 at 5 18 51 PM

**Comment by opr** [comment]
Ok, so it is being picked up by select2, and the `select2-selection--single` span has `aria-required="true"`. Not sure why your screen reader isn't voicing this while mine is. If you update the `aria-label` on that span, does the screen reader correctly read the updated text? Trying to narrow down whether this is the element it's reading from, or if it's looking at something else.

**Comment by frosso** [comment]
> If you update the aria-label on that span, does the screen reader correctly read the updated text? Trying to narrow down whether this is the element it's reading from, or if it's looking at something else.

Still no luck for my scenario, using VoiceOver on Safari.
https://github.com/woocommerce/woocommerce/assets/273592/fc51b96d-242e-4ac4-91e3-d43bcb0c2aee

I also have another recording with the devtools open on Safari, in case that helps. It's just a bit messier :D 

https://github.com/woocommerce/woocommerce/assets/273592/a7df26e6-f6f4-4854-ab11-6413d8a36faa

**Comment by frosso** [comment]
@opr thank you for the time you dedicated today. As we discussed, I made the following changes:
- Added the `aria-required` to the combobox/selectWoo
- Added `aria-hidden` on the `required` abbr

**Review by opr — CHANGES_REQUESTED** [review]
Hi - it looks like the changes to selectWoo.full.js aren't copied over so the country doesn't have the `aria-required` attribute.

**Comment by frosso** [comment]
@opr 🤦 I had it modified locally and didn't push it

Do you happen to know if I also need to modify `plugins/woocommerce/client/legacy/js/selectWoo/selectWoo.js` somehow? Or if I need to implement the changes in https://github.com/woocommerce/selectWoo?

**Comment by opr** [comment]
> @opr 🤦 I had it modified locally and didn't push it
> 
> Do you happen to know if I also need to modify `plugins/woocommerce/client/legacy/js/selectWoo/selectWoo.js` somehow? Or if I need to implement the changes in [woocommerce/selectWoo](https://github.com/woocommerce/selectWoo)?

I couldn't see any code path that loaded `plugins/woocommerce/client/legacy/js/selectWoo/selectWoo.js`  but maybe you want to do it just in case, with that said I didn't know about that selectWoo repository, makes sense that we should edit there as well. @barryhughes could you shed some light on the repository @frosso linked, is this the actual source for selectWoo and should we be editing the files there instead?

**Comment by barryhughes** [comment]
> @barryhughes could you shed some light on the repository @frosso linked, is this the actual source for selectWoo and should we be editing the files there instead?

Yes, in theory that is the actual source of SelectWoo. Unfortunately, I believe there may have been some divergence over time, with some edits being made directly to the copy *here,* but not always being applied over there (and possibly vice versa).

If we can also raise a PR making the same change over there, though, I think that would be beneficial. cc @naman03malhotra for any other thoughts, as you've recently been looking at Select2/Woo.

**Comment by naman03malhotra** [comment]
>, I believe there may have been some divergence over time, with some edits being made directly to the copy here, but not always being applied over there (and possibly vice versa).

If we were to make changes in the WooSelect repo and override the build file in core, we would lose the changes we make directly to the copy in core. So, we would need to do some further deep diving to make sure we have a common set of changes in the final result.

**Review by naman03malhotra — COMMENTED** [review]
>, I believe there may have been some divergence over time, with some edits being made directly to the copy here, but not always being applied over there (and possibly vice versa).

If we were to make changes in the WooSelect repo and override the build file in core, we would lose the changes we made directly to the copy in core. So, we would need to do some further deep diving to make sure we have a common set of changes in the final result and that maybe a tricky thing to do.

**Comment by frosso** [comment]
@barryhughes @naman03malhotra I just took a quick look - after all the moving around, [these are the latest changes](https://github.com/woocommerce/woocommerce/commits/37205ff4e0f489f28345895b12a0d5d602b0af4d/assets/js/selectWoo) to selectWoo in this repository.
Notice [the last commit](https://github.com/woocommerce/woocommerce/commit/41f63bd4877c5fd4f808b77eb7693ef08bc8c2d1) matches [the last commit in the selectWoo repository](https://github.com/woocommerce/selectWoo/commit/69d51b0624b2925a3c121d78b8e5d15d45bdb955) 3 years ago.

I created this PR on the selectWoo repository to basically mirror these changes: https://github.com/woocommerce/selectWoo/pull/53
I'm not sure who to assign it to 👀

**Comment by barryhughes** [comment]
Thanks, @frosso: @naman03malhotra I assigned to you (the usual round-robin doesn't seem to work in that repo, and I have low availability next week so couldn't grab it myself).

**Comment by frosso** [comment]
The addition of `aria-hidden` on the `abbr` would cause regressions across multiple repositories using e2e tests that run on multiple WC core versions.
We decided to defer this as a future enhancement: https://github.com/woocommerce/woocommerce/issues/48948

**Comment by barryhughes** [comment]
@frosso should https://github.com/woocommerce/selectWoo/pull/53 also be put on hold?

**Comment by frosso** [comment]
> @frosso should https://github.com/woocommerce/selectWoo/pull/53 also be put on hold?

@barryhughes sorry I missed your comment - I don't think there's anything that should hold it

**Review by naman03malhotra — COMMENTED** [review]
>I don't think there's anything that should hold it

@frosso I am still able to see the changed made to `select2` files, in this PR https://github.com/woocommerce/selectWoo/pull/53

Since we have now [removed](https://github.com/woocommerce/woocommerce/pull/48731) Select2 altother from core, probably we can also remove it from the SelectWoo repo to avoid confusion.

**Comment by frosso** [comment]
> @frosso I am still able to see the changed made to select2 files, in this PR https://github.com/woocommerce/selectWoo/pull/53
> Since we have now https://github.com/woocommerce/woocommerce/pull/48731 Select2 altother from core, probably we can also remove it from the SelectWoo repo to avoid confusion.

@naman03malhotra as it stands, the selectWoo repository still contains Select2.
How would you prefer to proceed? I have a few options:

1. First remove Select2 from [the SelectWoo repository](https://github.com/woocommerce/selectWoo), then update https://github.com/woocommerce/selectWoo/pull/53 accordingly
2. First merge https://github.com/woocommerce/selectWoo/pull/53 , then update the SelectWoo repository to remove select2

I made the changes to the selectWoo repository just as a mirror of the changes in the main WC Core repository.

Do you think this PR is currently on hold because select2 is not currently removed from the selectWoo repository?

**Review by naman03malhotra — APPROVED** [review]
Thank you for working on it, works well with the given testing instructions for checkout shortcode page.

image

**Review by naman03malhotra — COMMENTED** [review]
>Do you think this PR is currently on hold because select2 is not currently removed from the selectWoo repository?

Sorry for the confusion, I though [that PR](https://github.com/woocommerce/selectWoo/pull/53) and this are meant to go together, but I see they have separate concerns.

>as it stands, the selectWoo repository still contains Select2.
How would you prefer to proceed? I have a few options:
First remove Select2 from [the SelectWoo repository](https://github.com/woocommerce/selectWoo), then update https://github.com/woocommerce/selectWoo/pull/53 accordingly
First merge https://github.com/woocommerce/selectWoo/pull/53 , then update the SelectWoo repository to remove select2
I made the changes to the selectWoo repository just as a mirror of the changes in the main WC Core repository.

Either approach looks good to me, I'll let you decide how to proceed. https://github.com/woocommerce/selectWoo/pull/53 Looks good aparart from minor changes I've suggested.

**Review by opr — APPROVED** [review]
Re-tested, and it's working well for me, thanks @frosso I think rebasing should fix the issues with failing CI.

**Pull request merged**
